src/Container/Plugin/Guard/FinalizeGuardFactory.php

```diff
@@ -21,13 +22,58 @@
 final class FinalizeGuardFactory
 {
     /**
+     * @var bool
+     */
+    private $exposeEventMessageName;
+
+    /**
+     * FinalizeGuardFactory constructor.
+     * @param bool $exposeEventMessageName
+     */
+    public function __construct($exposeEventMessageName = false)
+    {
+        $this->exposeEventMessageName = $exposeEventMessageName;
+    }
+
+    /**
+     * Creates a new instance with exposeMessageName flag, specifically meant to be used as static factory.
+     *
+     * Configuration example:
+     *
+     * <code>
+     * <?php
+     * return [
+     *     \Prooph\ServiceBus\Plugin\Guard\FinalizeGuard::class => [
+     *         \Prooph\ServiceBus\Container\Plugin\Guard\FinalizeGuardFactory::class,
+     *         'exposeMessageName'
+     *     ]
+     * ];
+     * </code>
+     *
+     * @param string $name
+     * @param array $arguments
+     * @return \Prooph\ServiceBus\Plugin\Guard\FinalizeGuard
+     * @throws InvalidArgumentException
+     */
+    public static function __callStatic($name, array $arguments)
+    {
+        if (!isset($arguments[0]) || !$arguments[0] instanceof ContainerInterface) {
+            throw new InvalidArgumentException(
+                sprintf('The first argument must be of type %s', ContainerInterface::class)
+            );
+        }
+
+        return (new static(true))->__invoke($arguments[0]);
```

Comment:
ah __callStatic should be used when you want to expose message name, right? Because you always pass true as argument to __construct